app/cdap/components/GridTable/components/NestedMenu/index.tsx

```diff
@@ -72,29 +74,37 @@ export default function({
         }}
         className={classes.root}
       >
-        {menuOptions?.map((eachOption, index) => (
-          <MenuItemComponent
-            item={eachOption}
-            columnType={columnType.toLowerCase()}
-            index={index}
-            onMenuClick={handleMenuClick}
-          />
-        ))}
-        <MenuComponent
-          anchorEl={anchorEl2}
-          columnType={columnType.toLowerCase()}
-          menuOptions={nestedOptions}
-          setAnchorEl={setAnchorEl2}
-          submitOption={(e, item) => {
-            e.preventDefault();
-            e.stopPropagation();
-            setAnchorEl(null);
-            handleMenuOpenClose(title);
-            setAnchorEl2(null);
-            submitMenuOption(item.value, item.supported_dataType);
-          }}
-        />
+        {menuOptions?.map((eachOption, index) => {
+          return (
+            <MenuItemComponent
+              item={eachOption}
+              columnType={columnType.toLowerCase()}
+              index={index}
+              onMenuClick={handleMenuClick}
+            />
+          );
+        })}
+        {menuComponentOptions?.length > 0 &&
+          menuComponentOptions.map((options, index) => {
+            console.log('anchorEl', anchorEl);
+            return (
+              <MenuComponent
+                anchorEl={anchorEl?.length > 1 ? anchorEl[index + 1] : null}
+                columnType={columnType.toLowerCase()}
+                menuOptions={options}
+                setAnchorEl={setAnchorEl}
+                setMenuComponentOptions={setMenuComponentOptions}
+                submitOption={(e, item) => {
```

Comment:
can the item be renamed to clickedItem or selectedItem something based on when this event is triggered
Also, see e can be renamed according to the behavior of the event, onClickEvent or onSelectionEvent or onCheckedEvent like that, this give the reader an understanding of what kind of event is triggering this particular function

Reply:
Done